1.0/docs/devguide/templates.md

```diff
@@ -166,6 +168,79 @@ Changing a `manager.type` field should now cause the list to be re-sorted:
 
     this.set('employees.0.manager.type', 'engineer');
 
+#### Dynamic sort and filter changes
+
+The `observe` property lets you specify item sub-properties to 
+observe for filtering and sorting purposes. However, sometimes you want to
+dynamically change the sort or filter based on another unrelated value. In 
+this case, you can use a computed binding to _return_ a dynamic filter or 
+sort function when one or more dependent properties changes.
+
+    <dom-module id="employee-search">
+
+      <template>{% raw %}
+        <input value="{{searchString::change}}">
+        <template is="dom-repeat" items="{{employees}}" as="employee"
+            filter="{{matches(searchString)}}">
+            <div>{{employee.lastname}}, {{employee.firstname}}</div>
+        </template>{% endraw %}
+      </template>
+
+      <script>
+        Polymer({
+          is: "employee-search",
+          matches: function(string) {
+            if (!string) {
+              // if no search string, show all records
+              return function() { return true; }
+            } else {
+              // return a filter function for the current search string
+              string = string.toLowerCase();
+              return function(employee) {
+                var first = employee.firstname.toLowerCase();
+                var last = employee.lastname.toLowerCase();
+                return first.includes(string) ||
+                    last.includes(string);
+              }
+            }
+          },
+          properties: {
+            employees: {
+              type: Array,
+              value: function() {
+                return [
+                  { firstname: "Jack", lastname: "Aubrey" },
+                  { firstname: "Anne", lastname: "Elliot" },
+                  { firstname: "Stepehen", lastname: "Maturin" },
+                  { firstname: "Emma", lastname: "Woodhouse" }
+                ]
+              }
+            }
+          }
+        });
+      </script> 
+    </dom-module>
+
+#### Filtering on array index
+
+Because of the way {{site.project_title}} tracks arrays internally, the array 
+index isn't passed to the filter function. Looking up the array index for an
+item is an O(n) operation. Doing so  in a filter function could have 
+**significant performance impact**.
+
+If you want to look up the array index, you can use code like this:
+
+    filter: function(item) {
+      var index = this.items.indexOf(item);
```

Comment:
To be clear, this is precisely the costly `O(n)` operation referenced above;  as such, `filter` won't do this as a matter of course, but the user could decide to pay the price and write the code you have here.

As is, it sort-of reads that the example code is a way to _avoid_ the **"significant performance impact"**.  L231 could instead read something like "Nevertheless, if array index is required, it can be manually looked up using code like this:".  Slight nuance.